### PR TITLE
validation: timer: add new tests

### PR DESCRIPTION
**Pull request by TuomasTaipale**
Add new tests to check ability to create maximum amount of oneshot and periodic timer pools, maximum/large amount of timers from these pools and ability for implementations to handle timer pools with varying parameterization.

### CODE REVIEW

File: test/validation/api/timer/timer.c [hunk 1]
```diff
@@ -1465,6 +1465,80 @@ static void timer_pool_tick_info(void)
...
+{
+	odp_timer_capability_t capa;
+	odp_shm_t shm_tmr;
+	odp_timer_pool_t pool;
+	odp_timer_t *timer;
+	odp_timer_pool_param_t tp_param;
+	odp_queue_param_t queue_param;
+	odp_queue_t queue;
+	/* Use some sane value in case of memory-capped or very large max timer count. */
+	uint32_t num = MAX_TIMER_POOLS;
```
Comment (MatiasElo) [inline test/validation/api/timer/timer.c]: A bit confusing define to use here. There is also `NTIMERS` define or the default value could be simply defined here.

File: test/validation/api/timer/timer.c [hunk 2]
```diff
@@ -1465,6 +1465,80 @@ static void timer_pool_tick_info(void)
...
+	if (capa.queue_type_sched)
+		queue_param.type = ODP_QUEUE_TYPE_SCHED;
+
+	queue = odp_queue_create("test_max_alloc", &queue_param);
+
+	CU_ASSERT_FATAL(queue != ODP_QUEUE_INVALID);
+
+	odp_timer_pool_param_init(&tp_param);
+	tp_param.res_ns = res_ns;
+	tp_param.min_tmo = tmo_ns / 2;
```
Comment (MatiasElo) [inline test/validation/api/timer/timer.c]: Could check also `min_tmo` capability.

File: test/validation/api/timer/timer.c [hunk 3]
```diff
@@ -3255,6 +3329,92 @@ static void timer_test_periodic_pool_create_max(void)
...
+	odp_shm_t shm_tmr;
+	odp_timer_periodic_capability_t periodic_capa;
+	odp_timer_pool_param_t timer_param;
+	odp_timer_pool_t pool;
+	odp_timer_t *timer;
+	odp_queue_param_t queue_param;
+	odp_queue_t queue;
+	double min_freq, max_freq;
+	/* Use some sane value in case of very large max timer count. */
+	uint32_t num = MAX_TIMER_POOLS;
```
Comment (MatiasElo) [inline test/validation/api/timer/timer.c]: A bit confusing define to use here.

File: test/validation/api/timer/timer.c [hunk 4]
```diff
@@ -3255,6 +3329,92 @@ static void timer_test_periodic_pool_create_max(void)
...
+
+	CU_ASSERT_FATAL(pool != ODP_TIMER_POOL_INVALID);
+	CU_ASSERT_FATAL(odp_timer_pool_start_multi(&pool, 1) == 1);
+
+	odp_queue_param_init(&queue_param);
+
+	if (timer_capa.queue_type_sched)
+		queue_param.type = ODP_QUEUE_TYPE_SCHED;
+
+	queue = odp_queue_create("test_max_alloc", &queue_param);
```
Comment (MatiasElo) [inline test/validation/api/timer/timer.c]: Could have `CU_ASSERT_FATAL` check. Queue name could also be updated.

File: test/validation/api/timer/timer.c [hunk 5]
```diff
@@ -1465,6 +1465,139 @@ static void timer_pool_tick_info(void)
...
+	time_limit = odp_time_local_strict_ns() + res_ns * steps;
+
+	while (odp_time_local_strict_ns() < time_limit) {
+		ev = capa.queue_type_sched ? odp_schedule(NULL, ODP_SCHED_NO_WAIT) :
+					     odp_queue_deq(queue);
+
+		if (ev == ODP_EVENT_INVALID)
+			continue;
+
+		tmr = odp_timeout_timer(odp_timeout_from_event(ev));
```
Comment (MatiasElo) [inline test/validation/api/timer/timer.c:1583]: Could have an assert also for event type.

File: test/validation/api/timer/timer.c (hunk 1 above)
Comment (TuomasTaipale) [reply test/validation/api/timer/timer.c]: Yes, agree, first commit in v2 now removes `MAX_TIMER_POOLS` constant (as no more references to it) and uses `NTIMERS` instead.

File: test/validation/api/timer/timer.c (hunk 2 above)
Comment (TuomasTaipale) [reply test/validation/api/timer/timer.c]: There were a couple styles used to parameterize the pool, swapped to use the ones in `global_mem->param.*` in v2.

File: test/validation/api/timer/timer.c (hunk 4 above)
Comment (TuomasTaipale) [reply test/validation/api/timer/timer.c]: v2.

File: test/validation/api/timer/timer.c (hunk 5 above)
Comment (TuomasTaipale) [reply test/validation/api/timer/timer.c:1583]: v2.
